### ISSUE TITLE
Convert JS files to TS

### ISSUE THREAD

**Issue by Divide-By-0** [zkemail/zk-email-verify#71]
Files include:
- regex_to_circom/regex_to_dfa.js

**Comment by Tejasp2003** [comment]
i would like to work on this

**Comment by Divide-By-0** [comment]
@Tejasp2003 feel free to take it, and make a PR!

**Comment by AnmolSirola** [comment]
@Divide-By-0 Is anyone still working on this pr ?

**Comment by Divide-By-0** [comment]
> @Divide-By-0 Is anyone still working on this pr ?

Nope, you're free to and make a PR!

**Comment by SneakY-NickY** [comment]
Is this still a relevant issue? I would like to work on this.

**Comment by Divide-By-0** [comment]
Hey, you're free to take it. Lexical.js in zk regex is still JS.

**Comment by SneakY-NickY** [comment]
Great, could you please assign this issue to me?

**Comment by SoraSuegami** [comment]
Hi! I am one of the developers of zk-regex.
Thank you very much for your help!

Could you convert regex.js (modification of lexical.js) and gen_circom.js (modification of gen.py) in zk-regex to TS instead of JS files in zk-email-verify?
https://github.com/zkemail/zk-regex/tree/main/packages/compiler/src

We are moving all of the regex-related codes to zk-regex repo.

**Comment by SneakY-NickY** [comment]
Sure thing! Is the issue [https://github.com/zkemail/zk-regex/issues/18](url) related to this?

**Comment by Divide-By-0** [comment]
> Sure thing! Is the issue [zkemail/zk-regex#18](https://github.com/zkemail/zk-regex/issues/18) related to this?

Yes, that is the same issue (we should deprecate this current issue #71 in favor of that one, now that the regex code has moved).

**Comment by SneakY-NickY** [comment]
I left a comment in there so you can assign me to it.
